mypyc/emitmodule.py

```diff
@@ -217,16 +257,44 @@ def generate_c_for_modules(self) -> List[Tuple[str, str]]:
                 decls.emit_lines(*declaration.decl)
 
         if self.shared_lib_name:
+            self.generate_linking_struct(ext_declarations, emitter)
+
             self.generate_shared_lib_init(emitter)
 
         ext_declarations.emit_line('#endif')
         declarations.emit_line('#endif')
 
-        return file_contents + [('__native.c', ''.join(emitter.fragments)),
-                                ('__native_internal.h', ''.join(declarations.fragments)),
-                                ('__native.h', ''.join(ext_declarations.fragments)),
+        return file_contents + [('__native{}.c'.format(self.lib_suffix),
+                                 ''.join(emitter.fragments)),
+                                ('__native_internal{}.h'.format(self.lib_suffix),
+                                 ''.join(declarations.fragments)),
+                                ('__native{}.h'.format(self.lib_suffix),
+                                 ''.join(ext_declarations.fragments)),
                                 ]
 
+    def generate_linking_struct(self, decl_emitter: Emitter, code_emitter: Emitter) -> None:
```

Comment:
Add docstring. Document the linking table somewhere (both the purpose and the structure).